ADAL/tests/unit/ios/ADKeychainTokenCacheTests.m

```diff
@@ -232,7 +260,67 @@ - (void)testRemoveAllForClientId
     [self verifyCacheContainsItem:item4];
 }
 
-- (void)testRemoveAllForUserIdAndClientId
+- (void)testRemoveAllForUserIdAndClientId_whenUserIdNil_shouldReturnNo
+{
+    XCTAssertTrue([self count] == 0, "Start empty.");
+    
+    ADAuthenticationError *error;
+    XCTAssertNotNil([mStore allItems:&error]);
+    ADAssertNoError;
+    
+    //add two items with the same client ID and same user ID but differnet resource
+    ADTokenCacheItem *item1 = [self adCreateCacheItem:@"eric@contoso.com"];
```

Comment:
nit: if this code is the same for all tests, can we have a helper method to add those 4 items?

Reply:
sure!

Reply:
After double check, I find that "those 4 items" are not the same for different tests:

for `testRemoveAllForClientId`, they are items with same clientid but different userid;
for `testWipeAllItemsForUserId`, they are items with same userid but different clientid;
for `testRemoveAllForUserIdAndClientId`, they are mixed items with different clientId and differentuserid